tensorflow/stream_executor/gpu/asm_compiler.cc

```diff
@@ -331,4 +331,99 @@ port::StatusOr<std::vector<uint8>> BundleGpuAsm(
   return std::vector<uint8>(result_blob.begin(), result_blob.end());
 }
 
+static std::string findRocmExecutable(const std::string binary_relative_path,
+                                      const std::string rocm_root_dir) {
+  auto env = tensorflow::Env::Default();
+  std::string binary_path =
+      tensorflow::io::JoinPath(rocm_root_dir, binary_relative_path);
+  VLOG(2) << "Looking for " << binary_relative_path << " at " << rocm_root_dir;
+  if (!env->FileExists(binary_path).ok()) {
+    binary_path = "<" + binary_path + " - NOT FOUND>";
+  }
+  return binary_path;
+}
+
+port::StatusOr<std::vector<uint8>> BundleGpuAsm(
+    std::vector<HsacoImage> images, const std::string rocm_root_dir) {
+  std::string clang_offload_bundler_path =
+      findRocmExecutable("llvm/bin/clang-offload-bundler", rocm_root_dir);
+
+  // Initialise the "--inputs" / "--targets" arguments for the
+  // clang-offload-bundler with a dummy file / host target triple...
+  // clang-offload-bundler requires 1 and only 1 host target triple
+  std::ostringstream inputs_list;
+  std::ostringstream targets_list;
+
+  inputs_list << "/dev/null";
+  targets_list << "host-x86_64-unknown-linux";
```

Comment:
Just curious: Would this work on windows? And if not, is there a way to make it work? Not in this PR but in general.

Reply:
I do not know.

ROCm, and hence ROCm TF,  is currently only supported on the Linux platform, so I have had the need to look into whether or not this will work on Windows. The `clang-offload-bundler` utility being used here, is from the standard llvm install, so there is nothing ROCm specific about the utility itself.

Reply:
Thanks for the reply. Makes sense, lets leave it at linux for now then.

Reply:
`clang-offload-bundler` itself is a platform independent tool. It aggregates several binary objects for different architectures into one binary blob.

It does require a host-side object be inside the bundled blob. As we normally only produce GPU binaries in XLA or MLIR path, we populate a dummy host object just to tame `clang-offload-bundler`.

As ROCm runs strictly on the Linux platform, the current implementation from @deven-amd shall suffice for now.